### ISSUE TITLE
Selecting dead buffer problem

### ISSUE THREAD

**Issue by minad** [oantolin/embark#160]
I get somtimes caught in an unescapable selecting-dead-buffer problem.

1. `C-x b C-. L` (create live occur from buffers)
2. Switch to live occur window, I have the consult preview mode activated
3. Kill the live window
4. ESC ESC ESC
5. Every minibuffer command throws the error "Selecting dead buffer"

The problem seems to be this lambda:

~~~ elisp
             (:live
              (lambda ()
                (setf (buffer-local-value 'embark-collect-from buffer) nil)
                (with-current-buffer buffer
                  (save-match-data
                    (rename-buffer
                     (replace-regexp-in-string " Live" "" (buffer-name))
                     t)))
                (run-at-time 0 nil #'pop-to-buffer buffer)))
~~~

**Comment by minad** [comment]
It seems to be sufficient to guard everything with `buffer-live-p` as follows:

~~~ diff
-                (setf (buffer-local-value 'embark-collect-from buffer) nil)
-                (with-current-buffer buffer
-                  (save-match-data
-                    (rename-buffer
-                     (replace-regexp-in-string " Live" "" (buffer-name))
-                     t)))
-                (run-at-time 0 nil #'pop-to-buffer buffer)))
+                (when (buffer-live-p buffer)
+                  (with-current-buffer buffer
+                    (setq-local embark-collect-from nil)
+                    (save-match-data
+                      (rename-buffer
+                       (replace-regexp-in-string " Live" "" (buffer-name))
+                       t)))
+                  (run-at-time 0 nil #'pop-to-buffer buffer))))
~~~

**Comment by oantolin** [comment]
Thanks!

**Comment by oantolin** [comment]
I defensively added the liveness check in the snapshot case too, in case in the future we bring back the ability to create a snapshot without quitting the minibuffer.

**Comment by minad** [comment]
Thanks. 

> I defensively added the liveness check in the snapshot case too, in case in the future we bring back the ability to create a snapshot without quitting the minibuffer.

Yes, I thought about that too. But live occur is mostly good enough. One thing I dislike about the exiting snapshot is that no Consult preview is available anymore. I considered proposing to transfer the consult preview state to embark-consult such that preview could continue to work despite the minibuffer being closed. But I've quickly given up the thought since it would be a major complication for both Embark and Consult.

**Comment by oantolin** [comment]
Is MELPA getting faster? This change is already on there!

**Comment by minad** [comment]
It seems me to me that MELPA fluctuates heavily. I've observed that packages are already being published while the current build is still running. Maybe you just pushed your change in the middle of a running build. Originally I had expected that everything is done in atomic transactions, or at least the final publication step. The problem with the non-atomic publication is that if you are pushing changes to two packages in lock-step, for example selectrum+consult, consult gets updated earlier than selectrum and then you have a small window where you observe breakage. I saw that recently. But it is a small issue admittedly.
